src/NUnitFramework/framework/Internal/Builders/NUnitTestCaseBuilder.cs

```diff
@@ -61,7 +63,7 @@ public TestMethod BuildTestMethod(IMethodInfo method, Test parentSuite, TestCase
 
             CheckTestMethodSignature(testMethod, parms);
 
-            if (parms == null || parms.Arguments == null)
+            if (parms == null || parms.Arguments.Length == 0)
```

Comment:
Arguments can no longer be null as of https://github.com/nunit/nunit/pull/3488/commits/c3241d4df14f9a782b70ca8b2d736a2c5fb466e0 from #3488.

I don't know if it's the right change, but the reason I made the change was because TestCaseAttribute.GetParametersForTestCase makes the assumption that TestCaseParameters.Attributes (which implements ITestData.Attributes) is not null. Two TestCaseParameters constructors left it null and two didn't, and consistency felt right.

Reply:
It would be `params?.Arguments.Length ?? 0 == 0`, right? I can do that if you like it better.

Reply:
Doesn't matter enough to bother, just commenting 👍